projects/step-frontend/src/lib/os-plugins/modules/script-editor/components/script-editor/script-editor.component.ts

```diff
@@ -52,6 +57,14 @@ export class ScriptEditorComponent implements AfterViewInit, OnDestroy {
     this.saveInternal().subscribe(() => this._keywordExecutor.executeKeyword(this._functionId));
   }
 
+  canExit(): boolean {
+    if (this.noChanges) {
+      return true;
+    } else {
+      return confirm('You have unsaved changes. Do you want to navigate anyway?');
```

Comment:
@lukasz-lepa Please, Instead of browser's native confirm inject `DialogsService` and use `showWarning` method

Reply:
Done